### PR TITLE
fix regression for advTechTiles

### PR DESCRIPTION
**Pull request by rzulian**

### CODE REVIEW

**Comment by coyotte508** [comment]
Can you provide a test that is fixed by this?

This is the exact opposite of the last commit I made: https://github.com/donkeytech/gaia-engine/commit/f847de25a7883e872b312ba9e3e6d2c75b44ea18 (in which I edited a test too)

**Comment by rzulian** [comment]
Your test should fail, you cannot peek a adv tech tile and enter lev 5, unless you have 2 green Federations. On the other end, when you peek an adv tech tile, you can advance in any allowed research area.

File: src/tiles/techs.spec.ts [hunk 1]
```diff
@@ -102,13 +102,17 @@ describe("Tech Tiles", () => {
...
 
     // tslint:disable-next-line no-unused-expression
-    expect(engine.player(Player.Player1).data.advTechTiles.find(tile => tile.pos === AdvTechTilePos.GaiaProject)).to.not.be.undefined;
-    expect(engine.advTechTiles[AdvTechTilePos.GaiaProject].numTiles).to.equal(0);
+
+    expect(() => engine.move('p1 build lab -2x3. tech adv-gaia. cover free1. up gaia.')).to.throw();
+
+    const engine1 = new Engine(moves);
+    expect(() => engine1.move('p1 build lab -2x3. tech adv-gaia. cover free1. up terra.')).to.not.throw();
+    expect(engine1.player(Player.Player1).data.advTechTiles.find(tile => tile.pos === AdvTechTilePos.GaiaProject)).to.not.be.undefined;
```
Comment (coyotte508) [inline src/tiles/techs.spec.ts]: You should move `// tslint:disable-next-line no-unused-expression` to above this line (currently on line 109/110)

**Comment by coyotte508** [comment]
Your second commit is good, it fixes the green federation issue. The first commit is unnecessary. Adv tech tiles still could choose any field, since `'adv-gaia' !== 'gaia'`, so picking `adv-gaia` doesn't lock the player into advancing gaia track.

**Comment by rzulian** [comment]
AH boy, you are right ! I thought was 'gaia' === 'gaia', forcing to go to gaia! Anyways, should I reverse the commit?

**Comment by coyotte508** [comment]
You can do `git revert 9900859` and push :)

**Pull request merged**